simulation/traffic_simulator/test/src/behavior/test_longitudinal_speed_planner.cpp

```diff
@@ -0,0 +1,324 @@
+// Copyright 2015 TIER IV, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+
+#include <cmath>
+#include <geometry_msgs/msg/accel.hpp>
+#include <geometry_msgs/msg/twist.hpp>
+#include <geometry_msgs/msg/vector3.hpp>
+#include <traffic_simulator/behavior/longitudinal_speed_planning.hpp>
+
+#include "../expect_eq_macros.hpp"
+
+#define EXPECT_CONSTRAINTS_BOUNDED(DATA, lower, upper) \
+  EXPECT_GE(DATA.max_speed, lower);                    \
+  EXPECT_GE(DATA.max_acceleration, lower);             \
+  EXPECT_GE(DATA.max_deceleration, lower);             \
+  EXPECT_GE(DATA.max_acceleration_rate, lower);        \
+  EXPECT_GE(DATA.max_deceleration_rate, lower);        \
+  EXPECT_LE(DATA.max_speed, upper);                    \
+  EXPECT_LE(DATA.max_acceleration, upper);             \
+  EXPECT_LE(DATA.max_deceleration, upper);             \
+  EXPECT_LE(DATA.max_acceleration_rate, upper);        \
+  EXPECT_LE(DATA.max_deceleration_rate, upper);
+
+int main(int argc, char ** argv)
+{
+  testing::InitGoogleTest(&argc, argv);
+  return RUN_ALL_TESTS();
+}
+
+auto makeLongitudinalSpeedPlanner()
+  -> traffic_simulator::longitudinal_speed_planning::LongitudinalSpeedPlanner
+{
+  return traffic_simulator::longitudinal_speed_planning::LongitudinalSpeedPlanner{0.001, "entity"};
+}
+
+auto makeTwistWithLinearX(double x) -> geometry_msgs::msg::Twist
+{
+  return geometry_msgs::build<geometry_msgs::msg::Twist>()
+    .linear(geometry_msgs::build<geometry_msgs::msg::Vector3>().x(x).y(0.0).z(0.0))
+    .angular(geometry_msgs::build<geometry_msgs::msg::Vector3>().x(0.0).y(0.0).z(0.0));
+}
+
+auto makeAccelWithLinearX(double x) -> geometry_msgs::msg::Accel
+{
+  return geometry_msgs::build<geometry_msgs::msg::Accel>()
+    .linear(geometry_msgs::build<geometry_msgs::msg::Vector3>().x(x).y(0.0).z(0.0))
+    .angular(geometry_msgs::build<geometry_msgs::msg::Vector3>().x(0.0).y(0.0).z(0.0));
+}
+
+/**
+ * @note Test basic functionality. Test with an accelerating situation.
+ */
+TEST(LongitudinalSpeedPlanner, isAccelerating_true)
+{
+  const auto planner = makeLongitudinalSpeedPlanner();
```

Comment:
Almost all tests create an instance of LongitudinalSpeedPlanner, but this item can be reduced in description by using Gtest's [TestFixture](https://google.github.io/googletest/primer.html#same-data-multiple-tests). 
Please use it to reduce the amount of test code.